modules/javafx.controls/src/main/java/javafx/scene/control/skin/VirtualFlow.java

```diff
@@ -856,6 +856,8 @@ public String getName() {
 
         @Override protected void invalidated() {
             int cellCount = get();
+            resetSizeEstimates();
+            recalculateAndImproveEstimatedSize(2);
```

Comment:
We can use recalculateEstimatedSize() instead of this method.
The effect is the same improvement with a size of 2.

Reply:
True, good suggestion. I pushed that change.